src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -3392,6 +3371,43 @@ public boolean updateTypeForSyntheticClasses(
     throw new RuntimeException("Could not find the corresponding missing class!");
   }
 
+  /**
+   * Lookup the fully-qualified names of each type in the given string, and replace the simple type
+   * names in the given string with their fully-qualified equivalents. Return the result.
+   *
+   * @param javacType a type from javac
+   * @return that same type, with simple names in the class-to-package map replaced with FQNs
+   */
+  private String lookupFQNs(String javacType) {
+    // It's possible that the type could start with a new (synthetic) type variable's declaration.
+    // That won't be parseable as a type, so strip it first and then re-add it.
+    String typeVarDecl, rest;
+    if (javacType.startsWith("<")) {
```

Comment:
I am confused. How can a type start with `<`?

Reply:
Technically, it can't. However, we post-process what we get from javac to add synthetic type variable declarations to some return types (where they'll be placed in front of the method). So, for example, we might have something like `<SyntheticTypeVar> SyntheticTypeVar` as the input to this method; the first part is the declaration of the type variable, and the second part is a use of the type variable. (There's a test that demonstrates this - I think `LambdaBodyStaticUnsolved2Test`.)

I'll clarify the comment above.